end-to-end-test/tests/onboarding/onboarding.ts

```diff
@@ -0,0 +1,166 @@
+        }
+
+        await this.page.waitForSelector('#deriv_app > #app_contents > .dc-themed-scrollbars > form > .dc-btn--primary');
+        await this.page.click('#deriv_app > #app_contents > .dc-themed-scrollbars > form > .dc-btn--primary');
+    }
+
+    async updateServerURLAndAppIDInLocalStorage() {
+        const server = process.env.ENDPOINT;
+        const app_id = process.env.APPID;
+        await this?.signupPage?.evaluate(`localStorage.setItem('config.server_url', '${server}');`);
+        await this?.signupPage?.evaluate(`localStorage.setItem('config.app_id', '${app_id}');`);
+        await this?.signupPage?.evaluate(`window.location.reload();`);
+        const server_url = await this?.signupPage?.evaluate(() => {
+            const result = localStorage.getItem('config.server_url');
+            return Promise.resolve(result);
+        });
+        expect(server_url).toBe(process.env.ENDPOINT);
+        await suspend(10000);
+    }
+    async demoWizardHandler() {
+        await this.page.locator('.static-dashboard-wrapper__header > h2', { hasText: 'CFDs' });
+        await this.page.locator('.static-dashboard-wrapper__header > h2', { hasText: 'Multipliers' });
+        await this.page.locator('button[type="submit"]', { hasText: 'Next' }).click();
+        await this.page.locator('button[type="submit"]', { hasText: 'Next' }).click();
+        await this.page.locator('button[type="submit"]', { hasText: 'Next' }).click();
+        await this.page.locator('button[type="submit"]', { hasText: 'Next' }).click();
+    }
+    async cookieDialogHandler() {
+        if (this.page.locator('.cookie-banner'))
+            await this.page.locator('.cookie-banner > button[type=submit]', { hasText: /Accept/ }).click();
+    }
+    async signUp() {
+        await this.changeEndpoint();
+        await this.page.goto(process.env.APP_URL!);
+        await this.page.waitForSelector('#dt_signup_button');
+        const [newPage] = await Promise.all([
+            this.page.context().waitForEvent('page'), // get `context` by destructuring with `page` in the test params; 'page' is a built-in event, and **you must wait for this like this,**, or `newPage` will just be the response object, rather than an actual Playwright page object.
+            await this.page.click('#dt_signup_button'),
+        ]);
+        this.signupPage = newPage;
+        await suspend(10000);
+        await this.updateServerURLAndAppIDInLocalStorage();
+        await this.signupPage.waitForLoadState();
+        await this.signupPage.locator('input[name=email]#dm-email-input').isVisible();
+        await this.signupPage.locator('input[name=email]#dm-email-input').type(this.email);
+        process.env.email = this.email;
+        await this.signupPage.waitForSelector(
+            '.signup__Form-sc-1bdbun8-1 > ._signup-new__SignupContent-sc-1f1r3le-0 > label > .checkbox__CheckboxContainer-sc-r1zf4m-0 > .checkbox__StyledCheckbox-sc-r1zf4m-3'
```

Comment:
I'm not sure but AFAIK these generated hashed class names changes and when that happens your these tests will fail. is that correct?

Reply:
yess, you are right. i need to update them.
didn't notice that since they are auto generated

Reply:
done @shayan-deriv 
would you please check one more time?